todo/src/reducers/reducers.js

```diff
@@ -0,0 +1,73 @@
+import uuid from 'uuid';
+
+import {
+  ADD_TODO,
+  TOGGLE_COMPLETED,
+  DELETE_TODO,
+  DELETE_COMPLETED
+} from '../actions/actions';
+
+let initialState;
+const retrievedState = localStorage.getItem('storedStateTodoListRedux');
+const parsedState = JSON.parse(retrievedState);
+parsedState
+  ? (initialState = parsedState)
+  : (initialState = {
+      todos: [
+        {
+          id: 'be9a374e-0e79-42d7-ae9d-17ae542205ac',
+          value: 'sand the floor',
+          completed: false
+        },
+        {
+          id: '2a4add54-cbca-40ea-bb07-fa3e0b67618a',
+          value: 'paint the fence',
+          completed: false
+        },
+        {
+          id: 'e12a5b08-9ab3-4aab-805f-905c811e431b',
+          value: 'wax on',
+          completed: false
+        },
+        {
+          id: 'b5d28ffa-8b34-48b4-bbb6-a1085b90d04b',
+          value: 'wax off',
+          completed: false
+        }
+      ]
+    });
+
+const writeToLocalStorage = state => {
+  localStorage.setItem('storedStateTodoListRedux', JSON.stringify(state));
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_TODO:
+      let tempTodos = [
+        ...state.todos,
+        { id: uuid.v4(), value: action.payload, completed: false }
+      ];
+      writeToLocalStorage({ ...state, todos: tempTodos });
```

Comment:
ahhhhh, very clean way of accomplishing the localstorage! Another method I've seen is making TodoList a class component to get access to lifecycle hooks, use a componentWillUpdate and checked the nextProps against current props to see if they were new and set them to storage there -- this is much cleaner since it will only happen when the ADD_TODO action occurs :100: